### ISSUE TITLE
CASResult display issue when CASResult has DataFrame

### ISSUE THREAD

**Issue by kybowm** [sassoftware/python-swat#108]
For swat version 1.8.0, the print statement produces an error: 

``` python 
import swat
s=swat.CAS("cloud.example.com", 5570)
res=s.caslibinfo()
print(res)
```
> AttributeError: 'DataFrameFormatter' object has no attribute 'to_html'

Some Notes:
* The issue seems to be with any CASResult that should display a DataFrame
* I was able to produce the error in Jupyter and the Python interpreter.
* I was unable to produce the error using SAS Studio/CASL
* I was unable to produce the error with swat 1.8.1-dev (via interpreter)
* Swat 1.8.0 is currently the default package from PyPi and Conda Forge. 

Here is the full stack trace:

> ---------------------------------------------------------------------------
> AttributeError                            Traceback (most recent call last)
> ~\AppData\Local\Continuum\anaconda3\envs\DL2\lib\site-packages\IPython\core\formatters.py in __call__(self, obj)
>     700                 type_pprinters=self.type_printers,
>     701                 deferred_pprinters=self.deferred_printers)
> --> 702             printer.pretty(obj)
>     703             printer.flush()
>     704             return stream.getvalue()
> 
> ~\AppData\Local\Continuum\anaconda3\envs\DL2\lib\site-packages\IPython\lib\pretty.py in pretty(self, obj)
>     400                             meth = cls._repr_pretty_
>     401                             if callable(meth):
> --> 402                                 return meth(obj, self, cycle)
>     403                         if cls is not object \
>     404                                 and callable(cls.__dict__.get('__repr__')):
> 
> ~\AppData\Local\Continuum\anaconda3\envs\DL2\lib\site-packages\swat\cas\results.py in _repr_pretty_(self, p, cycle)
>     176             p.break_()
>     177             with p.indent(1):
> --> 178                 for line in pretty(item).splitlines():
>     179                     p.break_()
>     180                     p.text(line)
> 
> ~\AppData\Local\Continuum\anaconda3\envs\DL2\lib\site-packages\IPython\lib\pretty.py in pretty(obj, verbose, max_width, newline, max_seq_length)
>     140     stream = StringIO()
>     141     printer = RepresentationPrinter(stream, verbose, max_width, newline, max_seq_length=max_seq_length)
> --> 142     printer.pretty(obj)
>     143     printer.flush()
>     144     return stream.getvalue()
> 
> ~\AppData\Local\Continuum\anaconda3\envs\DL2\lib\site-packages\IPython\lib\pretty.py in pretty(self, obj)
>     400                             meth = cls._repr_pretty_
>     401                             if callable(meth):
> --> 402                                 return meth(obj, self, cycle)
>     403                         if cls is not object \
>     404                                 and callable(cls.__dict__.get('__repr__')):
> 
> ~\AppData\Local\Continuum\anaconda3\envs\DL2\lib\site-packages\swat\dataframe.py in _repr_pretty_(self, p, cycle)
>     638             p.text('...')
>     639             return
> --> 640         p.text(self.to_string())
>     641 
>     642     def to_string(self, apply_formats=None, **kwargs):
> 
> ~\AppData\Local\Continuum\anaconda3\envs\DL2\lib\site-packages\swat\dataframe.py in to_string(self, apply_formats, **kwargs)
>     680         formatter.columns = formatter.tr_frame.columns
>     681 
> --> 682         txt = formatter.to_string()
>     683         if txt is None:
>     684             if getattr(formatter, 'buf', None) is not None:
> 
> AttributeError: 'DataFrameFormatter' object has no attribute 'to_string'

> ---------------------------------------------------------------------------
> AttributeError                            Traceback (most recent call last)
> ~\AppData\Local\Continuum\anaconda3\envs\DL2\lib\site-packages\IPython\core\formatters.py in __call__(self, obj)
>     343             method = get_real_method(obj, self.print_method)
>     344             if method is not None:
> --> 345                 return method()
>     346             return None
>     347         else:
> 
> ~\AppData\Local\Continuum\anaconda3\envs\DL2\lib\site-packages\swat\cas\results.py in _repr_html_(self)
>     114             output.append('<div class="cas-results-body">')
>     115             if hasattr(item, '_repr_html_'):
> --> 116                 res = item._repr_html_()
>     117                 if res is None:
>     118                     output.append('<div>%s</div>' % res)
> 
> ~\AppData\Local\Continuum\anaconda3\envs\DL2\lib\site-packages\swat\dataframe.py in _repr_html_(self)
>     725             #       formatters on a DataFrame that is truncated in the console.
>     726             formatter.columns = formatter.tr_frame.columns
> --> 727             html = formatter.to_html(**notebook_opts)
>     728             if html is None:
>     729                 if getattr(formatter, 'buf', None) is not None:
> 
> AttributeError: 'DataFrameFormatter' object has no attribute 'to_html'

**Comment by kesmit13** [comment]
Can you tell me what version of pandas you have installed?

**Comment by kybowm** [comment]
Pandas 1.2.0. I didn't explicitly install it, so it was part of the dependencies for either swat or dlpy.

**Comment by kesmit13** [comment]
SWAT doesn't do its own rendering of DataFrames. It farms that off to pandas. However, the pandas library has been moving those functions around from version to version which makes it very difficult to keep downstream libraries like SWAT stable. You might want to pin your pandas version to 1.1 or 1.0 when installing SWAT for now. I'll look into what's different about pandas 1.2.

**Comment by kybowm** [comment]
That makes sense. Using pandas version 1.1 works. I'm closing this issue, but I think without some guidance first time installers will run into the same issue. I don't know if you can force pandas < 1.2 at installation or not. If nothing else, we could add a line on the README page.

Thanks for the help!

**Comment by kesmit13** [comment]
Fixed in v1.8.1

**Comment by kybowm** [comment]
Verified. Thanks for moving so quickly!
